includes/multi-currency/class-settings.php

```diff
@@ -0,0 +1,338 @@
+	 * @return array
+	 */
+	public function get_settings( $current_section = '' ) {
+		$settings = [];
+
+		if ( '' === $current_section ) {
+			$settings = apply_filters(
+				$this->id . '_enabled_currencies_settings',
+				[
+					[
+						'title' => __( 'Enabled currencies', 'woocommerce-payments' ),
+						// TODO: Learn more link needed.
+						'desc'  => sprintf(
+							/* translators: %s: url to documentation. */
+							__( 'Accept payments in multiple currencies. Prices are converted based on exchange rates and rounding rules, and include conversion fees. <a href="%s">Learn more</a>', 'woocommerce-payments' ),
+							''
+						),
+						'type'  => 'title',
+						'id'    => $this->id . '_enabled_currencies',
+					],
+
+					[
+						'type' => 'wcpay_multi_currencies',
+					],
+
+					[
+						'type' => 'sectionend',
+						'id'   => $this->id . '_enabled_currencies',
+					],
+				]
+			);
+		} elseif ( 'store' === $current_section ) {
+			$settings = apply_filters(
+				$this->id . 'store_settings',
+				[
+					[
+						'title' => __( 'Store settings', 'woocommerce-payments' ),
+						// TODO: Need learn more link.
+						'desc'  => sprintf(
+							/* translators: %s: url to documentation. */
+							__( 'Store settings allow your customers to choose which currency they would like to use when shopping at your store. <a href="%s">Learn more</a>', 'woocommerce-payments' ),
+							''
+						),
+						'type'  => 'title',
+						'id'    => $this->id . '_store_settings',
+					],
+
+					[
+						'title'         => __( 'Store settings', 'woocommerce-payments' ),
+						'desc'          => __( 'Automatically switch customers to the local currency if it is enabled above.', 'woocommerce-payments' ),
+						// TODO: Need preview link... what is expected?
+						'desc_tip'      => sprintf(
+							/* translators: %s: url to preview? */
+							__( 'Customers will be notified via store alert banner. <a href="%s">Preview</a>', 'woocommerce-payments' ),
+							''
+						),
+						'id'            => $this->id . '_enable_auto_currency',
+						'default'       => 'no',
+						'type'          => 'checkbox',
+						'checkboxgroup' => 'start',
+					],
+
+					[
+						'desc'          => __( 'Add a currency switcher to the cart widget', 'woocommerce-payments' ),
+						// TODO: Need configure link... what is expected? Widgets?
+						'desc_tip'      => sprintf(
+							/* translators: %s: url to the widgets? */
+							__( 'A currency switcher is also available in your widgets. <a href="%s">Configure now</a>', 'woocommerce-payments' ),
+							''
+						),
+						'id'            => $this->id . '_enable_cart_switcher',
+						'default'       => 'no',
+						'type'          => 'checkbox',
+						'checkboxgroup' => 'end',
+					],
+
+					[
+						'type' => 'sectionend',
+						'id'   => $this->id . 'store_settings',
+					],
+				]
+			);
+		} else {
+			foreach ( WC_Payments_Multi_Currency()->get_enabled_currencies() as $currency ) {
```

Comment:
Perhaps we should consider initialising an instance of `WC_Payments_Multi_Currency()` in the constructor for the Settings class so that we can reuse the same instance here and also in the `get_currency_setting` function to save us running the Multi Currency setup code multiple times.

Reply:
Makes sense, and good catch. Addressed this in https://github.com/Automattic/woocommerce-payments/pull/2007